core/shared/src/main/scala/scalaz/zio/IOQueue.scala

```diff
@@ -85,7 +85,7 @@ class IOQueue[A] private (capacity: Int, ref: IORef[State[A]]) {
   final def interruptTake[E](t: Throwable): IO[E, Boolean] =
     IO.flatten(ref.modifyFold[E, IO[E, Boolean]] {
       case Deficit(takers) if takers.nonEmpty =>
-        val forked: IO[E, Unit] = IO.forkAll(takers.toList.map(_.interrupt[E](t).toUnit))
+        val forked: IO[E, Unit] = IO.forkAll(takers.toList.map(_.interrupt[E](t))).toUnit
```

Comment:
In this and the other change, place a TODO comment to use the composite Fiber. Doesn’t have to be done now but will be useful.